src/coord/src/coord.rs

```diff
@@ -189,6 +188,31 @@ pub struct Coordinator {
     /// A map from connection ID to metadata about that connection for all
     // active connections.
     active_conns: HashMap<u32, ConnMeta>,
+
+    // We need to track various frontiers.
+    // 1) `indexes` tracks the `upper` and computed `since` of the
+    // indexes. The `since` is the time at which we are willing to compact
+    // up to. `determine_timestamp()` uses this as part of its heuristic when
+    // determining a viable timestamp for queries.
+    // 2) `txn_reads` tracks active read transactions so that we don't compact any
+    // indexes beyond an in-progress transaction.
+    // 3) `compaction_frontiers` is a merged version of `indexes` and `txn_reads`
+    // that is used to inform the dataflow workers what they should compact.
+    // 4) `since_updates` holds pending compaction messages to be sent to the
+    // dataflow workers.
+    //
+    // How this should always work is that `determine_timestamp` will always
+    // return the least valid since of a query. Thus, any new transactions should
+    // always be >= the current compaction frontier and so should never change the
+    // frontier when being added to `compaction_frontiers`. The compaction frontier
+    // may change when a transaction ends (if it was the oldest transaction and
+    // the index's since was advanced after the transaction started) or when
+    // `update_upper` is run (if there are no in progress transactions before
+    // the new since). When it does, it is added to `since_updates` and will be
+    // processed during the next `maintenance` call.
```

Comment:
I'm a bit weirded out by this commentary in the middle of a struct definition. Is there a better place to land module-level documentation about transactions?